akka-actor/src/main/scala/akka/routing/TailChopping.scala

```diff
@@ -83,8 +88,14 @@ private[akka] final case class TailChoppingRoutees(
 }
 
 /**
- * A router poll that sends the message to a first, random picked, routee,
- * then wait a specified `interval` and then send to a second, random picked, and so on till one full cycle..
+ * A router pool with retry logic, intended for cases where a return message is expected in
+ * response to a message sent to the routee. As each message is sent to the routing pool, the
+ * routees are randomly ordered. The message is sent to the first routee. If no response is received
+ * before the `interval` has passed, the same message is sent to the next routee. This process repeats
+ * until either a response is received from some routee, the routees in the pool are exhausted, or
+ * the `within` duration has passed since the first send. If no routee sends
+ * a response in time, a [[akka.actor.Status.Failure]] wrapping a [[akka.pattern.AskTimeoutException]]
+ * is sent to the sender.
```

Comment:
Good improvements, however it does not explain the *why* we're doing this.

Let's also add something like this while we're fixing docs here:

```
The goal of this router is to decrease tail latencies ("chop off the tail", hence the name)
in cases where multiple routees can perform the same piece of work, and may sometimes
act much slower than usually. In this case, sending the additional work requests (also 
known as "backup requests") allows to decrease total response time considerably 
as it's less probably that multiple workers are expiriencing slowness at the same time.

This technique is explained in depth in Jeff Dean's presentation on 
<a href="http://static.googleusercontent.com/media/research.google.com/en//people/jeff/Berkeley-Latency-Mar2012.pdf">Achieving Rapid Response Times in Large Online Services</a>.
```